transformer_engine/pytorch/tensor/_internal/float8_tensor_base.py

```diff
@@ -100,7 +107,10 @@ def get_metadata(self) -> Dict[str, Any]:
 
     def prepare_for_saving(self) -> Tuple[list[Optional[torch.Tensor]], Float8TensorBase]:
         """Prepare the tensor base for saving for backward"""
-        tensors = [self._data, self._transpose]
+        tensors = [self._data, self._transpose, self._scale_inv]
+        self._data = None
+        self._transpose = None
+        self._scale_inv = None
```

Comment:
We've gone back and forth a few times about whether to clear the data when saving (added in https://github.com/NVIDIA/TransformerEngine/pull/1500, removed in https://github.com/NVIDIA/TransformerEngine/pull/1548). Based on the last discussion (https://github.com/NVIDIA/TransformerEngine/pull/1548#pullrequestreview-2668006078, https://github.com/NVIDIA/TransformerEngine/pull/1548#issuecomment-2707658650), making sure that the weight cache produces PyTorch tensors should fix the bugs we were seeing before. Pinging @ksivaman and @ptrendx for a sanity check.

Reply:
I am aware of this - this PR came from my discussion with Pawel on Monday :-). But I agree - we need to check this carefully with multiple scenarios. Maybe @guyueh1 could help run the NeMo test suite with this change?